### DISCUSSION TITLE
renovate-runner is not working with Gradle variables

### DISCUSSION THREAD

**Discussion by ikercrg** [renovatebot/renovate#8872] [Request Help] [answered]
**What Renovate type, platform and version are you using?**

I use the one which is explained here: [https://gitlab.com/renovate-bot/renovate-runner](https://gitlab.com/renovate-bot/renovate-runner)

"renovateVersion": "24.63.0"

Our company has its own GitLab instance and I use a mirror of the renovate/runner project for creating the project that is mentioned.

**Describe the bug**

Renovate is working fine except the part related to Gradle. 

**Relevant debug logs**

WARN: Gradle extraction failed (repository=pd-de/document-server/document-server)
       "errMessage": "Command failed: ./gradlew --init-script renovate-plugin.gradle renovate\nWARNING: An illegal reflective access operation has occurred\nWARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v7.Java7$1 (file:/home/ubuntu/.gradle/wrapper/dists/gradle-6.0-bin/2i8td48mp24knj9qorsduq0p9/gradle-6.0/lib/groovy-all-1.3-2.5.8.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)\nWARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v7.Java7$1\nWARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations\nWARNING: All illegal access operations will be denied in a future release\n\nFAILURE: Build failed with an exception.\n\n* Where:\nSettings file '/builds/pd-de/document-server/renovate-runner/renovate/repos/gitlab/pd-de/document-server/document-server/settings.gradle' line: 9\n\n* What went wrong:\nA problem occurred evaluating settings 'document-server'.\n> **Could not get unknown property 'artifactory_url' for object of type** org.gradle.api.internal.artifacts.repositories.DefaultMavenArtifactRepository.\n\n* Try:\nRun with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.\n\n* Get more help at https://help.gradle.org\n\nBUILD FAILED in 6s\n"

**Additional context**

The problem is clear, because in the build.gradle are a lot of references to  artifactory_url, artifactory_user and artifactory_api_token. Something like the following:

`pluginManagement {
    repositories {
        maven {
            name 'CaptainMavenAll'
            url "${artifactory_url}/pd_de-maven2-all"
            credentials {
                username = "${artifactory_user}"
                password = "${artifactory_api_token}"
            }
        }
    }
}`

Probably this is not a bug, but the problem is that I do not know how to follow because I get always the same error.  I tried adding a renovate.json with trustLevel=high, copying the Gradle variables (ORG_GRADLE_PROJECT_xxx) to new variables, using the property hostRules without success. Is there any chance to solve this issue that I get?

Thanks in advance (great project BTW)

**Comment by rarkins** [comment, marked as answer]
Can you check in case the "gradle-lite" manager already satisfies your use cases? It's a JS parser so avoids needing to run Gradle for reasons just like you're describing. 

"gradle-lite": {
  "enabled": true
}

You can also disable gradle while you're at it using similar syntax

**Reply by ikercrg** [reply]
It works! Thank you!!
